RecoTracker/LST/plugins/alpaka/LSTModulesDevESProducer.cc

```diff
@@ -0,0 +1,54 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "RecoTracker/LST/interface/LSTModulesDev.h"
+#include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
+
+// LST includes
+#include <SDL/ModuleMethods.h>
+#include <SDL/LST.h>
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  
+class LSTModulesDevESProducer : public ESProducer {
+public:
+  LSTModulesDevESProducer(const edm::ParameterSet &iConfig);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+  std::optional<SDL::modulesBuffer<alpaka_common::DevHost>> produce(const TrackerRecoGeometryRecord &iRecord);
+
+private:
+  const std::string txtFile_;
+};
+
+LSTModulesDevESProducer::LSTModulesDevESProducer(const edm::ParameterSet &iConfig)
+  : ESProducer(iConfig), txtFile_{iConfig.getParameter<edm::FileInPath>("txt").fullPath()}
+{
+  setWhatProduced(this, iConfig.getParameter<std::string>("ComponentName"));
+}
+
+void LSTModulesDevESProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("ComponentName", "")->setComment("Product label");
+  desc.add<edm::FileInPath>("txt", edm::FileInPath())
+      ->setComment("Path to the txt file for the module map parameters");
+  descriptions.addWithDefaultLabel(desc);
+}
+
+  std::optional<SDL::modulesBuffer<alpaka_common::DevHost>> LSTModulesDevESProducer::produce(const TrackerRecoGeometryRecord &iRecord) {
+    // write directly to SDL : FIXME : SHOULD NOT HAPPEN HERE
+    SDL::modulesBuffer<alpaka_common::DevHost> modules(cms::alpakatools::host());
+    alpaka::QueueCpuBlocking queue(cms::alpakatools::host());
+    SDL::LST::loadMaps();
+    SDL::loadModulesFromFile(&modules, SDL::nModules, SDL::nLowerModules, *SDL::pixelMapping, queue, txtFile_.c_str());
```

Comment:
(possibly only just curious) For what is the CPU `queue` used here?

Reply:
The queue is used in e.g. `alpaka::memcpy(queue, modulesBuf->nModules_buf, src_view_nModules);`

Reply:
In general we have `struct modulesBuffer<TAcc>* modulesBuf` and the method is used to fill the GPU modules as well (in the standalone version).
In this specific case in the ESProducer, things are explicitly on host/CPU.

So, I'm not sure how to interpret your suggestion to use `std::memcpy` in this combined context.

Reply:
Ok, I understand the code is more general towards the target device of that `memcpy()`, and the host target is just a special case. Thanks for the clarification. Would a comment be worth it?

Reply:
> Would a comment be worth it?

sure, I can add